include/ginkgo/core/factorization/par_ilu.hpp

```diff
@@ -157,7 +171,17 @@ class ParIlu : public Composition<ValueType> {
         : Composition<ValueType>(factory->get_executor()),
           parameters_{factory->get_parameters()}
     {
-        generate_l_u(system_matrix, parameters_.skip_sorting)->move_to(this);
+        if (parameters_.l_strategy == nullptr) {
+            parameters_.l_strategy =
+                std::make_shared<typename l_matrix_type::classical>();
+        }
+        if (parameters_.u_strategy == nullptr) {
+            parameters_.u_strategy =
+                std::make_shared<typename u_matrix_type::classical>();
+        }
```

Comment:
Could you also check whether l_strategy and u_strategy are different?

Reply:
Doesn't this problem appear only for automatical?

Reply:
yes, it only for automatical.

Reply:
I think `automatical` should be fixed in a new small PR, by always creating clones when appropriate. Hopefully, there is a way to do this without propagating that `automatical` internal issue everywhere in Ginkgo.

Reply:
I see

Reply:
Yes, I would also fix the `automatical` issue in another PR where we do that in a central place (for example in the constructor of `Csr`).